src/mon/OSDMonitor.cc

```diff
@@ -5797,6 +5797,35 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
+int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, ostream *ss)
+{
+  int64_t max_pgs_per_osd = g_conf->mon_pg_warn_max_per_osd;
+  int64_t max_pgs = max_pgs_per_osd * osdmap.get_num_in_osds();
```

Comment:
the 'make check' result suggests we should use MAX(num_in_osds, 1).

for the mon_pg_warn_max_per_osd == 0 case, which someone might have used to disable the warning, maybe we should default to 200 instead, since there's a nice message explaining it if that limit is hit here.

Reply:
I used max with 3, since that seems like the minimum useful cluster size.  (And it fixes the standalone tests, which create pools before any osds exist).  Sound ok?

Reply:
Yeah we can do that.  In that case I think we should rename the config option to be more like mon_max_pgs_per_osd (remove the 'warn' part)?

Reply:
Yeah, 3 seems fine, and renaming the config option is a good idea too.